cpm

```diff
@@ -315,22 +315,26 @@ _guix() {
     esac
 }
 
- _nix() {
-     case "$OP" in
-         install) nix-env -iA "$@";;
-         remove)  niv-env -e "$@";;
-         list)    nix-env -q "$@";;
-         count)   nix-env -q | wc -l;;
-         update)  nix-channel --update;;
-         upgrade) nix-env -u;;
-         search)  nix-env -qa "$@";;
-         show)    nix-env -qa --description "$@";;
-         f*)      pem "unsupported: this feature is functionally useless in this PM";;
-         clean)   nix-collect-garbage -d;;
-     esac
- }
+_nix() {
+    case "$OP" in
+        install) nix-env -iA "$@";;
+        remove)  niv-env -e "$@";;
+        list)    nix-env -q "$@";;
+        count)   nix-env -q | wc -l;;
+        update)  nix-channel --update;;
+        upgrade) nix-env -u;;
+        search)  nix-env -qa "$@";;
+        show)    nix-env -qa --description "$@";;
+        f*)      pem "unsupported: this feature is functionally useless in this PM";;
+        clean)   nix-collect-garbage -d;;
+    esac
+}
 
-if   has apk; then
+# Use _pm=FUNCTION cpm COMAMND to force a specyfic cmp function
+# ie.: _pm=_portage cpm list
+if [ -n "$_pm" ]; then
```

Comment:
Personally I'd do

```
if [ "$_pm" ]; then
```

But that's just personal preference (the `-n` but isn't needed)